### DISCUSSION TITLE
Non-fretted (non-guitar) instruments transposing for standard notation

### DISCUSSION THREAD

**Discussion by mxAlexG** [CoderLine/alphaTab#916] [Q&A] [answered]
Hello,
Judging by what I understood from [issue#139 ](https://github.com/CoderLine/alphaTab/issues/139) and the behavior I am getting, alphaTab renders non-guitar tracks in standard notation an octave lower than they are in the rendered tablature bars because guitars (including bass guitar) are written one octave higher than sounding pitch. An example would be like the following:
 
NON-GUITAR INSTRUMENTS:
Tablature: G5 (G String - 12th fret)  -->  Standard Notation: G4   --> displayValue(midi note): G4 (67)

GUITAR INSTRUMENTS:
Tablature: G5 (G String - 12th fret)  -->  Standard Notation: G5   --> displayValue(midi note): G5 (79)

Is this correct?

My guess is that alphaTabs performs this octave lower transposing step based on the midi program numbers? or some value that is relevant to GuitarPro file format spec?

What is the best way to tell which track will be identified as a guitar instrument that will get this octave lower transposing step?

Thanks.

**Comment by Danielku15** [comment, marked as answer]
Guitar Pro 6-8 files have this transposition encoded inside the data model. So it is Guitar Pro who decides there which tracks are visually transposed. In alphaTab this is handled as `Staff.displayTranspositionPitch` 

https://github.com/CoderLine/alphaTab/blob/f8da16bd56ef46269fc1bc48142afa985d0af6dd/src/importer/GpifParser.ts#L1045-L1075

In alphaTex with explicit tunings the transposition is disabled for 3 specific tunings ([see also](https://alphatab.net/docs/alphatex/metadata#tuning)): 
https://github.com/CoderLine/alphaTab/blob/f8da16bd56ef46269fc1bc48142afa985d0af6dd/src/importer/AlphaTexImporter.ts#L718-L721

In alphaTex for automatic tunings it is aligned a bit more with Guitar Pro (I analyzed there once all instrument types manually in GP):
https://github.com/CoderLine/alphaTab/blob/f8da16bd56ef46269fc1bc48142afa985d0af6dd/src/importer/AlphaTexImporter.ts#L1003-L1041

For Guitar Pro 3-5 the logic is that the display is transposed for all [Midi Guitar](https://github.com/CoderLine/alphaTab/blob/f8da16bd56ef46269fc1bc48142afa985d0af6dd/src/midi/GeneralMidi.ts#L52-L53) tracks only: 
https://github.com/CoderLine/alphaTab/blob/f8da16bd56ef46269fc1bc48142afa985d0af6dd/src/importer/Gp3To5Importer.ts#L377-L378

So it is not 100% consistent across all file format and would maybe need some alignment. But that's how it is done currently. 


https://music.stackexchange.com/questions/101432/why-is-guitar-music-written-one-octave-higher
https://hubguitar.com/sight-reading/octaves-and-transposition

**Comment by mxAlexG** [comment]
Hi,
Thanks for the thorough answer. I appreciate it.
